### PR TITLE
refactor: bash scripts

### PR DESCRIPTION
**Pull request by septs**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Setup Toolchain
-        run: ./scripts/setup-debian.sh
+        run: sudo ./scripts/setup-debian.sh ${{ matrix.build.variant }}
```
Comment (CoelacanthusHex) [inline .github/workflows/build.yaml:32]: Why？

File: scripts/build-ci.sh [hunk 2]
```diff
@@ -1,66 +1,25 @@
...
-        ;;
-    *)
-        echo "$SAVED_PATH"
-        ;;
-    esac
-}
+source "$SCRIPT_DIR/functions.sh"
 
-set -x
+set -xeuo pipefail
```
Comment (CoelacanthusHex) [inline scripts/build-ci.sh:6]: `set -x` will pollute output, use it only when debugging.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (septs) [reply .github/workflows/build.yaml:32]: the script internal use `sudo apt-get`, this is move `sudo` to the script external.

File: scripts/build-ci.sh (hunk 2 above)
Comment (septs) [reply scripts/build-ci.sh:6]: i want the user to know which commands were run

File: .github/workflows/build.yaml (hunk 1 above)
Comment (CoelacanthusHex) [reply .github/workflows/build.yaml:32]: AFAIK, `sudo` is useless in  GitHub Actions, it's enough to just run `apt-get`.

File: scripts/build-ci.sh (hunk 2 above)
Comment (CoelacanthusHex) [reply scripts/build-ci.sh:6]: The user should read the script, instead of DoS the brain of maintainers.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (septs) [reply .github/workflows/build.yaml:32]: OK, i try it

File: scripts/build-ci.sh (hunk 2 above)
Comment (septs) [reply scripts/build-ci.sh:6]: OK

File: .github/workflows/build.yaml (hunk 1 above)
Comment (septs) [reply .github/workflows/build.yaml:32]: CleanShot 2024-06-21 at 13 49 22@2x

**Pull request merged**